### PR TITLE
feature(MTTL-2075): Checklist, Box components

### PR DESCRIPTION
**Pull request by kymmeh**

### CODE REVIEW

File: lib/components/box/box.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React, { forwardRef } from "react";
```
Comment (cloudratha) [inline lib/components/box/box.tsx:1]: Could almost argue that we could just extend Card to support these layouts?

File: lib/components/details/details.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import "./styles.scss";
+
+interface DetailsProps {
+  title: string;
+  children: ReactElement;
+  className?: string;
+}
+
+export const Details = ({ title, children, className }: DetailsProps) => {
```
Comment (cloudratha) [inline lib/components/details/details.tsx:14]: If you look at some of the other components I've tried to make all the common components support forwarding refs to functional components. ErrorSummary might be a good one to look at

File: lib/components/status-checklist/status-checklist.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        strokeLinecap="square"
+      />
+    </svg>
+  );
+};
+
+interface StatusChecklistProps {
+  items: {
+    label: string;
+    status: boolean;
```
Comment (cloudratha) [inline lib/components/status-checklist/status-checklist.tsx:46]: Status feels wrong here for a boolean. I would change status to something like `success` or something that indicates a boolean. Maybe also consider the name of the component, maybe drop the `Status` bit? Happy to discuss

File: lib/components/details/details.tsx (hunk 2 above)
Comment (kymmeh) [reply lib/components/details/details.tsx:14]: The only reason I didn't for this component is that it specifically uses "details" as it's root element; though if we wanted we could wrap that in a component?

**Review by cloudratha — APPROVED** [review]

**Pull request merged**